announcement/announcement-api/api/src/java/org/sakaiproject/announcement/api/AnnouncementsUserNotificationHandler.java

```diff
@@ -96,8 +100,9 @@ public Optional<List<UserNotificationData>> handleEvent(Event e) {
         // TODO: the following code could be simplified. Lots of try catches.
         try {
             // If the announcement has just been hidden or removed, remove any existing alerts for it
+            // Also if it has been saved as draft or with release date in the future
             if ((AnnouncementService.SECURE_ANNC_REMOVE_OWN.equals(e.getEvent()) || AnnouncementService.SECURE_ANNC_REMOVE_ANY.equals(e.getEvent()))
-                        || (AnnouncementService.EVENT_ANNC_UPDATE_AVAILABILITY.equals(e.getEvent()) && message.getHeader().getDraft())) {
+                    || (updateEvent.equals(e.getEvent()) && (message.getHeader().getDraft() || Instant.now().isBefore(message.getHeader().getInstant())))) {
```

Comment:
Do it clearer and put comments.

boolean isDraft = xxxx
boolean happensInTheFuture = yyyyy

&& (isDraft || happensInTheFuture)